SharedUtils/src/main/java/com/braintreepayments/api/sharedutils/GzipRequestInterceptor.kt

```diff
@@ -0,0 +1,55 @@
+package com.braintreepayments.api.sharedutils
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.asResponseBody
+import okio.GzipSource
+import okio.buffer
+
+/**
+ * An OkHttp interceptor that automatically decompresses gzip-encoded HTTP responses.
+ *
+ * This interceptor checks for responses with "Content-Encoding: gzip" header and
+ * transparently decompresses the response body. After decompression, it removes
+ * the Content-Encoding and Content-Length headers since they no longer apply
+ * to the decompressed content.
+ *
+ * The interceptor handles case-insensitive matching for the gzip encoding
+ * (e.g., "gzip", "GZIP", "Gzip" are all supported).
+ */
+internal class GzipRequestInterceptor : Interceptor {
```

Comment:
Did a bit of reading on this since we don't do this on iOS. Seems like the iOS URLSession handles it automatically as of iOS 7+. Pretty interesting that okhttp doesn't do the same but finding a lot that indicates it may require a more manual implementation on Android.

Reply:
Plus less code is always a win 😄